### PR TITLE
Implements first complete set of cb_* functions for Cohort Browser endpoints

### PR DESCRIPTION
**Pull request by sk-sahu**
> Hold on! Yes there are lot of **Files changed**. But don't worry half of them auto-generated HTML docs (starts with `./docs/`). But yeah even apart from that also many changes done to standardise few classes and their methods.

## Improvements  

* cloudos-class
* cohort-class

Creates a S4 class named `cloudos` (same as package name, this nomenclature inspired from [AnnotationHub package](https://github.com/Bioconductor/AnnotationHub/) ) with a complementary method. This will create a main object, which will hold many information related to CloudOS server including base URL, authentication and team ID.  Use of this object will simplify calling every other functions which requires to communicate with server.

Likewise, a S4 class named `cohort` for holding cohort related objects and methods.

## Adds few more functions

- connect_cloudos() | Connect CloudOS
- cb_create_cohort() | Create Cohort
- cb_extract_samples() | Extract participants
- cb_filter_metadata() | Filter metadata
- cb_filter_participants() | Filter participants
- cb_genotypic_save() | Genotypic save
- cb_get_cohort_filters() | Get cohort filters
- cb_get_filter_statistics() | Filter a cohort samples
- cb_get_genotypic_table() | Get genotypic table
- cb_get_samples_table() | Get samples table
- cb_list_cohorts() | List cohorts
- cb_load_cohort() | Get cohort information
- cb_plot_filters() | Plot filters
- cb_search_phenotypic_filters() | List available filters 

Also, some internal functions (unexposed) and Improvements to existing ones.

Check [README of this branch](https://github.com/lifebit-ai/cloudos/tree/feature-filter-api#cloudos-) for details with examples.

## Logo

Yeah, Now we have the typical hexagon R-package logo - [cloudos R-package logo](https://github.com/lifebit-ai/cloudos/blob/feature-filter-api/logo/hexlogo.png) This is very basic one, let me know if we can improve this one.

### CODE REVIEW

**Comment by sk-sahu** [comment]
As we (@Vlad-Dembrovskyi , @aldembert and me) had a discussion last Friday, most of them are same but major changes based on @Vlad-Dembrovskyi's input are - 
* A cohort object, which can be passed to all cohort related functions.
* Get samples table for a particular cohort with applied filters.

**Review by cgpu — CHANGES_REQUESTED** [review]
We need:

- unit tests for the functions
- reproducible dummy examples so that we can internally at a first instance test

I am suggesting to hold off merging until we have tests and examples, unless there is an urgent need to merge.

**Comment by sk-sahu** [comment]
Include UX feedbacks - https://app.zenhub.com/workspaces/lifebit-platform-5aa6276b1224bb03d13f0e82/issues/lifebit-ai/lifebit-platform/10156

File: R/cb_plots.R [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  }
+  all_values <- c()
+  for(i in 1:length(fields$field$values)){
+    all_values <- c(all_values, fields$field$values[[i]])
+  }
+  # generate the colour vector
+  my_value_color <- c()
+  for(i in all_values){
+      if(i %in% my_values){
+        my_value_color <- c(my_value_color, "green")
```
Comment (cgpu) [inline R/cb_plots.R:84]: Could use the hex codes from the web app for extra style points :sparkles:

File: R/create_cohort.R [hunk 2]
```diff
@@ -1,47 +0,0 @@
-#' @title Create Cohort
```
Comment (cgpu) [inline R/create_cohort.R:1]: Replaced by `cb_`

**Comment by sk-sahu** [comment]
@cgpu Just added `pkgdown.yaml` so that we can safely clean this `docs` folder (which is becoming lot of files to keep track in commits and during review). 

From github-actions, It will push the master branch push pkgdown updates to a new `gh-branch`

**Review by cgpu — APPROVED** [review]
Good to go! :neckbeard: 

Squash it, and right after create a new branch named `plots-improvements` so that we don't forget about those.

**Pull request merged**